src/hooks/appAction.ts

```diff
@@ -2,23 +2,39 @@ import { List } from 'immutable';
 import { QueryClient, useQuery } from '@tanstack/react-query';
 import * as Api from '../api';
 import { buildAppActionsKey } from '../config/keys';
-import { getApiHost, getData, getDataOrThrow } from '../config/utils';
+import { getApiHost, getData, getDataOrThrow, getPermissionLevel } from '../config/utils';
 import { QueryClientConfig } from '../types';
-import { convertJs } from '@graasp/sdk';
+import { PermissionLevel, convertJs } from '@graasp/sdk';
 import { AppActionRecord } from '@graasp/sdk/frontend';
 
-export default (queryClient: QueryClient, queryConfig: QueryClientConfig) => {
+export default (
+  queryClient: QueryClient,
+  queryConfig: QueryClientConfig,
+  useAppActionsUpdates?: (itemId: string | null | undefined) => void,
```

Comment:
I agree with your comment. It makes sense to split the factory and configure the hooks separately. While I kept everything in the same file at the moment (to avoid refactoring the tests yet), I changed that in [f642d60](https://github.com/graasp/graasp-apps-query-client/commit/f642d60d750a14f60bd057d80102b9d26b28611f)